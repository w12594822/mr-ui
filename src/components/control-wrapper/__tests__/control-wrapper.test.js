import React from 'react';
import { shallow } from 'enzyme';
import { testCases } from './control-wrapper-test-cases';

describe('ControlWrapper', () => {
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.basicWithError.description, () => {
    beforeEach(() => {
      testCase = testCases.basicWithError;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(testCases.jsxError.description, () => {
    beforeEach(() => {
      testCase = testCases.jsxError;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
