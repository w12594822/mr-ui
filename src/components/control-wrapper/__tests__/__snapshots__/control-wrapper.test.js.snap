// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlWrapper JSX error renders 1`] = `
<div
  className="relative "
>
  <span>
    fake input
  </span>
  <div
    role="alert"
  >
    <div
      className="bg txt-s mt3 py6 px12 bg-red-faint round border border--red color-red"
      data-test="control-wrapper-error"
      id="control-wrapper-error"
    >
      <div>
        <div
          className="txt-bold mb3"
        >
          Something went wrong!
        </div>
        <div>
          You made a huge mistake
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ControlWrapper basic renders 1`] = `
<div
  className="relative "
>
  <span>
    fake input
  </span>
  <div
    role="alert"
  />
</div>
`;

exports[`ControlWrapper basic with error renders 1`] = `
<div
  className="relative "
>
  <span>
    fake input
  </span>
  <div
    role="alert"
  >
    <div
      className="bg txt-s mt3 py6 px12 bg-red-faint round border border--red color-red"
      data-test="control-wrapper-error"
      id="control-wrapper-error"
    >
      There was an error.
    </div>
  </div>
</div>
`;
